app/src/main/java/app/revanced/manager/ui/screen/InstallerScreen.kt

```diff
@@ -69,6 +71,24 @@ fun InstallerScreen(
             onConfirm = { vm.install(it) }
         )
 
+    if (vm.showInstallerStatusDialog)
+        InstallerStatusDialog(object : InstallerStatusDialogModel() {
+            override var packageInstallerResult = vm.packageInstallerResult!!
+            override var showInstallerStatusDialog = vm.showInstallerStatusDialog
+                set(value) {
+                    field = value
+                    vm.showInstallerStatusDialog = value
+                }
+
+            override fun reinstall() {
+                vm.reinstall()
+            }
+
+            override fun install() {
+                vm.install(InstallType.DEFAULT) // Always default for now.
+            }
+        })
```

Comment:
This isn't great because the initialization code will run every time the screen recomposes.
Use the view model (preferred) or `remember` to avoid it.

Reply:
> Use the view model

What do you mean by that? Should I move line 75-90 to a method in the view model?

Reply:
Yes. Move the anonymous `InstallerStatusDialogModel` object to the viewmodel or make the viewmodel implement `InstallerStatusDialogModel`.